src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/SessionEvents.cs

```diff
@@ -36,10 +39,86 @@ public SessionStartEventArgs() : this(new DataCollectionContext(new SessionId(Gu
         public SessionStartEventArgs(DataCollectionContext context)
             : base(context)
         {
+            this.Properties = new Dictionary<string, object>();
             Debug.Assert(!context.HasTestCase, "Session event has test a case context");
         }
 
         #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Returns the Session Start Properties currently specified.
+        /// </summary>
+        public IEnumerator<KeyValuePair<string, object>> GetProperties()
```

Comment:
@singhsarab @vagisha-nidhi Should GetProperties be not part of DataCollectionEventArgs? In this way, we can add properties bag in other event args as well. For now they will be empty.

Reply:
As confirmed with @singhsarab, this can be in `SessionStartEventArgs`. We can port it to base class whenever the need arises.